tests/sources/rest_api/configurations/source_configs.py

```diff
@@ -395,6 +395,19 @@ def repositories():
             repositories(),
         ],
     },
+    {
+        "client": {
+            "base_url": "https://test",
+        },
+        "resources": [
+            {
+                "name": "test",
+                "client": {
+                    "auth": HttpBasicAuth("", "BASIC_AUTH_TOKEN"),
```

Comment:
example of a valid config with an `auth` override, such as what is needed for solving the issue described in the pull request.